lib/fhir_client/client.rb

```diff
@@ -101,23 +91,17 @@ def use_representation_preference
       @return_preference = FHIR::Formats::ReturnPreferences::REPRESENTATION
     end
 
-    def versioned_resource_class(klass)
-      if @fhir_version == :stu3
-        FHIR.const_get(klass)
-      else
-        FHIR::DSTU2.const_get(klass)
-      end
-    end
-
     def detect_version
       cap = capability_statement
-      if cap.is_a?(FHIR::CapabilityStatement)
-        @fhir_version = :stu3
+      if cap.is_a?(FHIR::STU3::CapabilityStatement) || cap.is_a?(FHIR::CapabilityStatement)
```

Comment:
Why not a separate `elsif cap.is_a?(FHIR::CapabilityStatement)`.... Seems weird to combine them in an `or` and then have the result be a ternary operation (i.e. another in-line `if`) ?

Reply:
Good catch, this can be changed because `try_conformance_formats` catches the differences.  I'll also change from setting the @fhir_version directly to using the `use_r4` method to ensure the formats are also updated.